src/lighteval/metrics/metrics.py

```diff
@@ -644,3 +650,45 @@ def all_metrics():
                 continue
             res.extend(as_list(metric.value.metric_name))
         return res
+
+    @staticmethod
+    def loglikelihood_acc_metric(normalization: Normalization | None = None) -> SampleLevelMetric:
+        """
+        Function for creating custom loglikelihood accuracy metric in runtime.
+        """
+
+        normalization_str = normalization_name(normalization) if normalization else ""
+        metric_name = f"acc_{normalization_str}"
+        return SampleLevelMetric(
+            metric_name=metric_name,
+            sample_level_fn=LoglikelihoodAcc(normalization=normalization).compute,
+            category=MetricCategory.MULTICHOICE if not normalization == PMINorm() else MetricCategory.MULTICHOICE_PMI,
+            use_case=MetricUseCase.ACCURACY,
+            corpus_level_fn=np.mean,
+            higher_is_better=True,
+        )
+
+    @staticmethod
+    def probability_metric(
+        normalization: Normalization | None = None,
+        return_mass: bool = False,
+        aggregation_function: Callable[[np.ndarray], float] = np.max,
+    ) -> SampleLevelMetric:
+        """
+        Function for creating custom probability metric in runtime.
+        """
+
+        mass_str = "mass" if return_mass else ""
+        normalization_str = normalization_name(normalization) if normalization else ""
+        metric_name = "_".join(filter(None, ["prob", mass_str, normalization_str]))
+
+        return SampleLevelMetric(
+            metric_name=metric_name,
+            sample_level_fn=Probability(
```

Comment:
This does not go here - it has nothing to do in the enum class, and it's not even needed since these functions are static

Reply:
Wdym by other dtailed metrics files ?
I too wasn't sure where to put this, but essentially there are full fledged metrics (just need few args to initialize), I don't something like this already in the codebase.
I placed them here because indeed the enum already have few staticmethods.

ANyways I guess I can create a new file dynamic|partial_metrics.py wdyt ?

Reply:
dynamic_metrics works for me!